roles/privacyidea_ha/tasks/generate_keys.yml

```diff
@@ -0,0 +1,50 @@
+
+---
+- name: Copy privacyidea config file 'pi.cfg'
+  template:
+    src: pi.cfg.j2
+    dest: /etc/privacyidea/pi.cfg
+  notify: restart uwsgi
+
+# Audit Keys (private.pem and public.pem) are used to sign and verify the audit logs, 
+# ensuring that the audit log entries are tamper-proof and can be verified.
+- name: Generate private key for PrivacyIDEA audit log
+  command: "{{ privacyidea_dir }}/virtualenv/bin/pi-manage create_audit_keys"
+  args:
+    chdir: "{{ privacyidea_dir }}"
+    creates: /etc/privacyidea/private.pem
+  become: true
+  when: is_main_node
+
+- name: Get auth private key 
+  ansible.builtin.fetch:
+    src: "/etc/privacyidea/private.pem"
+    dest: "/tmp/certificate_sync_key_{{ unique_ha_group }}.auth.priv"
+    flat: true
+  when: is_main_node
+
+- name: Get auth public key 
+  ansible.builtin.fetch:
+    src: "/etc/privacyidea/public.pem"
+    dest: "/tmp/certificate_sync_key_{{ unique_ha_group }}.auth.pub"
+    flat: true
+  when: is_main_node
+
+- name: Set up auth private key
+  ansible.builtin.copy:
+    dest: "/etc/privacyidea/private.pem"
```

Comment:
When Ansible downloads sensitive files locally we should use an Ansible block to delete the local files afterwards.

Reply:
Added block